optimade/server/config.py

```diff
@@ -110,6 +110,31 @@ class ServerConfig(BaseSettings):
         description="Absolute path to a JSON file containing the MongoDB collection of /links resources for the index meta-database",
     )
 
+    @validator("implementation", pre=True)
+    def set_implementation_defaults(cls, v):
+        """Set defaults and modify by passed value(s)"""
+        res = {
+            "name": "Optimade Python Tools",
+            "version": __version__,
+            "source_url": "https://github.com/Materials-Consortia/optimade-python-tools",
+            "maintainer": {"email": "dev@optimade.org"},
+        }
+        res.update(v)
+        return res
+
+    @validator("provider", pre=True)
+    def set_provider_defaults(cls, v):
+        """Set defaults and modify by passed value(s)"""
+        res = {
+            "prefix": "exmpl",
+            "name": "Example provider",
+            "description": "Provider used for examples, not to be assigned to a real database",
+            "homepage": "https://example.com",
+            "index_base_url": "http://localhost:5001",
+        }
+        res.update(v)
+        return res
+
```

Comment:
```suggestion
```
I don't think we should set defaults for the provider this way. Either the user provides the whole `Provider` model or None, which is the current behavior.